src/main/java/seedu/address/model/module/Semester.java

```diff
@@ -1,45 +1,59 @@
 package seedu.address.model.module;
 
 import java.util.ArrayList;
-
-import org.json.simple.JSONArray;
-import org.json.simple.JSONObject;
+import java.util.List;
 
 /**
  * The semester of the module
  */
 public class Semester {
-    private final String semesterNo;
-    private final ArrayList<Timetable> timetable = new ArrayList<>();
-
-    public Semester(JSONObject obj) {
-        this.semesterNo = obj.get("semester").toString();
-        JSONArray arr = (JSONArray) obj.get("timetable");
-
-        int i;
-        for (i = 0; i < arr.size(); i++) {
-            timetable.add(new Timetable((JSONObject) arr.get(i)));
-        }
+    private final SemesterNo semesterNo;
+    private final ArrayList<Lesson> timetable = new ArrayList<>();
+    private final Exam exam;
+
+    public Semester(SemesterNo semesterNo, List<Lesson> timetable,
+                    Exam exam) {
+        this.semesterNo = semesterNo;
+        this.exam = exam;
+        this.timetable.addAll(timetable);
     }
 
-    public Timetable getTimetable(String classNo) {
-        int i;
-        for (i = 0; i < timetable.size(); i++) {
-            if (timetable.get(i).getClassNo().equals(classNo)) {
-                return timetable.get(i);
+    /**
+     * Find all Lessons with the given lesson number string.
+     * @param lessonNo Lesson number string.
+     * @return ArrayList of Lessons matching the lesson number, returns empty list if no matching lessons.
+     */
+    public ArrayList<Lesson> findLessons(LessonNo lessonNo) {
+        ArrayList<Lesson> lessons = new ArrayList<>();
+        for (int i = 0; i < timetable.size(); i++) {
+            if (timetable.get(i).getLessonNo().equals(lessonNo)) {
+                lessons.add(timetable.get(i));
             }
```

Comment:
Consider using a `foreach` style loop to iterate over the `timetable` to increase code clarity.